raster-foundry/raster-foundry
# Add project layer mosaic endpoints

## Overview

This PR adds the following endpoints:
  - Get Layer Mosaic Definition                                                                                                                                                                                      
  - Get Scene Layer Color Corrections                                                                                                                                                                                
  - Update Scene Layer Color Corrections                                                                                                                                                                             
  - Bulk Update Scene Layer Color Corrections                                                                                                                                                                        
  - Add Scene Layer Order Endpoint 


### Checklist

- ~Styleguide updated, if necessary~
- ~Symlinks from new migrations present or corrected for any new migrations~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
- ~Any new SQL strings have tests~


## Testing Instructions

 * Make sure the following requests work:
    - `GET` to `api/projects/<projectId>/layers/<layerId>/mosaic`
    - `GET` to `api/projects/<projectId>/layers/<layerId>/mosaic/<sceneId>`
    - `PUT` to `api/projects/<projectId>/layers/<layerId>/mosaic/<sceneId>`
    - `POST` to `api/projects/<projectId>/layers/<layerId>/mosaic/bulk-update-color-corrections`
    - `PUT` to `api/projects/<projectId>/layers/<layerId>/order`
 * Change one or more digits in `<layerId>` so that the layer does not belong to the project -- the above requests should return `403`s. Changing `<projectId>` should be similar.

Closes https://github.com/raster-foundry/raster-foundry/issues/4521
